tfexec/terraform.go

```diff
@@ -0,0 +1,606 @@
+package tfexec
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+type Terraform struct {
+	execPath    string
+	workingDir  string
+	execVersion string
+	env         []string
+	logger      *log.Logger
+}
+
+// NewTerraform returns a Terraform struct with default values for all fields.
+// If a blank execPath is supplied, NewTerraform will attempt to locate an
+// appropriate binary on the system PATH.
+func NewTerraform(workingDir string, execPath string) (*Terraform, error) {
```

Comment:
this can just be `New`

Reply:
As above, can probably rename `workingDir` to `rootModulePath` or something to be more explicit?

Reply:
I think in this case we don't know if the path is actually going to even have any Terraform code? So should we be making assumption it's a root module? `terraform fmt` for example can run from anywhere, when the file content is piped into STDIN.

Reply:
Agree with Radek about the root module thing, working dir makes sense, but I still think this should just be `tfexec.New`.